pennylane/ops/functions/is_commuting.py

```diff
@@ -19,6 +19,46 @@
 from pennylane import numpy as np
 
 
+def _get_target_name(op):
+    """Get the name for the target operation. If the operation is not controlled, this is
+    simplify the operation's name.
+    """
+    _control_base_map = {
+        "CNOT": "PauliX",
+        "CZ": "PauliZ",
+        "CY": "PauliY",
+        "CSWAP": "SWAP",
+        "Toffoli": "PauliX",
+        "ControlledPhaseShift": "PhaseShift",
+        "CRX": "RX",
+        "CRY": "RY",
+        "CRZ": "RZ",
+        "CRot": "Rot",
+        "MultiControlledX": "PauliX",
+    }
+    if op.name in _control_base_map:
+        return _control_base_map[op.name]
+    if isinstance(op, qml.ops.op_math.Controlled):  # pylint: disable=no-member
+        return op.base.name
+    if op.name == "ControlledOperation":
+        return op.control_base
+    return op.name
+
+
+def _check_mat_commutation(op1, op2):
+    """Uses matrices and matrix multiplication to determine whether op1 and op2 commute.
+
+    ``op1`` and ``op2`` must be on the same wires.
+    """
+    op1_mat = op1.matrix()
+    op2_mat = op2.matrix()
+
+    mat_12 = np.matmul(op1_mat, op2_mat)
+    mat_21 = np.matmul(op2_mat, op1_mat)
+
+    return qml.math.allclose(mat_12, mat_21)
+
+
 def _create_commute_function():
```

Comment:
Why do we define `_commutes = _create_commute_function()`? Why not change this function to:
```suggestion
def _commutes(op_name1, op_name2):
```
And remove the `commutes_inner` function and return directly 
```
return op_name1 in commutation_map[op_name2]
```

Reply:
This was something I did last time I tried to improve this file.  I wanted to create the `commutation_map` dictionary once, but avoid sticking it in the global scope for the file.

This way, `commutation_map` is a variable in `_commutes`'s "enclosing scope".  It's created once and only accessible from the `_commutes` function.

But I'm not particularly tied to that decision.

Reply:
We can always just leave the `commutation_map` in the global file namespace.

Reply:
Ahh okay now I understand! Then this is fine don't worry.